pkg/rbd/nodeserver.go

```diff
@@ -148,24 +164,39 @@ func (ns *NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 	return &csi.NodeStageVolumeResponse{}, nil
 }
 
-func (ns *NodeServer) cleanupStagingPath(stagingTargetPath, devicePath, volID string, isStagePathCreated, isBlock, isMounted bool) {
+func (ns *NodeServer) undoStagingTransaction(stagingParentPath, stagingTargetPath, devicePath, volID string, isStagePathCreated, isMounted bool) {
 	var err error
+
 	if isMounted {
 		err = ns.mounter.Unmount(stagingTargetPath)
 		if err != nil {
 			klog.Errorf("failed to unmount stagingtargetPath: %s with error: %v", stagingTargetPath, err)
+			return
 		}
 	}
+
 	// remove the block file created on staging path
-	if isBlock && isStagePathCreated {
+	if isStagePathCreated {
 		err = os.Remove(stagingTargetPath)
 		if err != nil {
 			klog.Errorf("failed to remove stagingtargetPath: %s with error: %v", stagingTargetPath, err)
+			return
```

Comment:
this failure can leave stable rbd  devices, as there won't be any node unstage call if node staging fails

Reply:
Hmmm... ok we can remove this return. I am assuming you are fine with the other returns in this function, the reasoning for the same was, an umount failure will result in an unmap failure as the device would be busy anyway.

Reply:
fixed.